pilot/pkg/model/authentication.go

```diff
@@ -32,7 +31,7 @@ var JwtKeyResolver = newJwksResolver(JwtPubKeyExpireDuration, JwtPubKeyEvictionD
 // and/or service annotation. Once these legacy flags/config deprecated,
 // this function can be placed by a call to store.AuthenticationPolicyByDestination
 // directly.
-func GetConsolidateAuthenticationPolicy(mesh *meshconfig.MeshConfig, store IstioConfigStore, hostname Hostname, port *Port) *authn.Policy {
+func GetConsolidateAuthenticationPolicy(store IstioConfigStore, hostname Hostname, port *Port) *authn.Policy {
 	config := store.AuthenticationPolicyByDestination(hostname, port)
 	if config != nil {
```

Comment:
What about changing the logic to 

if config == nil {
  return nil
}
everything else...

Reply:
There is also a branch for error in JwtKeyResolver flow, so it's simpler to have `return nil` as a common default fallback.